includes/class-wc-subscriptions-core-plugin.php

```diff
@@ -447,6 +449,58 @@ public function register_post_statuses() {
 		}
 	}
 
+	/**
+	 * Filter the query_vars of a wc_get_orders() query to map 'any' to be all valid subscription statuses instead of
+	 * defaulting to only valid order statuses.
+	 *
+	 * @param $query_vars
+	 *
+	 * @return mixed
+	 */
+	public function map_order_query_args_for_subscriptions( $query_vars ) {
+		if ( ! wcs_is_custom_order_tables_usage_enabled() ) {
+			return $query_vars;
+		}
+
+		/**
+		 * Map the 'any' status to wcs_get_subscription_statuses() in HPOS environments.
+		 *
+		 * In HPOS environments, the 'any' status now maps to wc_get_order_statuses() statuses. Whereas, in
+		 * WP Post architecture 'any' meant any status except for ‘inherit’, ‘trash’ and ‘auto-draft’.
+		 *
+		 * If we're querying for subscriptions, we need to map 'any' to be all valid subscription statuses otherwise it would just search for order statuses.
+		 */
+		if ( isset( $query_vars['post_type'] ) && '' !== $query_vars['post_type'] ) {
+			// OrdersTableQuery::maybe_remap_args() will overwrite `type` with the `post_type` value.
+			if ( 'shop_subscription' !== $query_vars['post_type'] ) {
+				return $query_vars;
+			}
+
+			// Simplify the type logic.
+			$query_vars['type'] = 'shop_subscription';
+			unset( $query_vars['post_type'] );
+		}
+
+		if ( isset( $query_vars['type'] ) && 'shop_subscription' === $query_vars['type'] ) {
+			if ( isset( $query_vars['post_status'] ) && '' !== $query_vars['post_status'] ) {
+				// OrdersTableQuery::maybe_remap_args() will overwrite `status` with the `post_status` value.
+				if ( 'any' !== $query_vars['post_status'] ) {
```

Comment:
Question: do we need to worry about `$query_vars['post_status']` being an array of string `'any'` is is `post_status` always singular?

Reply:
Good catch.  

I went ahead and updated this to check against the converted array instead.